crates/apub/src/activities/following/follow.rs

```diff
@@ -91,6 +90,7 @@ impl ActivityHandler for Follow {
 
   #[tracing::instrument(skip_all)]
   async fn receive(self, context: &Data<LemmyContext>) -> Result<(), LemmyError> {
+    insert_received_activity(&self.id, context).await?;
```

Comment:
The bug is here, `insert_received_activity()` would be called once when the follow is received, then later again for the undo follow with `self.object.verify(context).await?;` which has the same id and fails with a db unique error. Moving  `insert_received_activity()` to receive fixes the problem, because it isnt called from UndoFollow.

As a precaution Ive also moved `insert_received_activity()`  into receive in all other activity handlers.